homeassistant/components/roborock/switch.py

```diff
@@ -75,15 +77,26 @@ async def async_setup_entry(
     coordinators: dict[str, RoborockDataUpdateCoordinator] = hass.data[DOMAIN][
         config_entry.entry_id
     ]
+    possible_entities = [
+        RoborockSwitchEntity(
+            f"{description.key}_{slugify(device_id)}",
+            coordinator,
+            description,
+        )
+        for device_id, coordinator in coordinators.items()
+        for description in SWITCH_DESCRIPTIONS
+    ]
+    # We need to check if this function is supported by the device.
+    results = await asyncio.gather(
+        *(entity.entity_description.get_value(entity) for entity in possible_entities),
+        return_exceptions=True,
+    )
+
     async_add_entities(
         (
-            RoborockSwitchEntity(
-                f"{description.key}_{slugify(device_id)}",
-                coordinator,
-                description,
-            )
-            for device_id, coordinator in coordinators.items()
-            for description in SWITCH_DESCRIPTIONS
+            entity
+            for entity, result in zip(possible_entities, results)
+            if not isinstance(result, HomeAssistantError)
```

Comment:
Should we blow up on other exceptions and not hide them?

Reply:
I would say probably yes? I'm not sure what the best way to do this is though. Just do another loop through all of results and if any are an Exception that isn't a RoborockException, then raise it?

Reply:
I'd do one loop and if there's another exception then raise it, if there's the expected exception then continue, and if there's no exception then add the entity to a list.